### ISSUE TITLE
Review and fix possible concurrency issues with the admission webhook

### ISSUE THREAD

**Issue by almaslennikov** [Mellanox/network-operator#714]
**What would you like to be added**:
The current implementation of the admission webhook might have issues when run concurrently. Need to review the code and make sure it's safe
**Why is this needed**:
If two or more validation requests are issued at the same time for different objs, the behavior may be undefined

**Comment by killianmuldoon** [comment]
I wrote a test to show the issue in this at https://github.com/Mellanox/network-operator/pull/715

This could be a pretty bad bug IMO for any user who's doing many updates. Are other webhooks in the same form? i.e. holding state in the webhooks themselves?

**Comment by adrianchiris** [comment]
thx to @ykulazhenkov  for digging through controller runtime code !

tl;dr using the default validator there is no issue, obj is copied per req, using custom validator, obj is not copied

pasting his findings:

for default implementation (like in kubebuilder book) this is not a problem, they copy Validator instance on each call
https://github.com/kubernetes-sigs/controller-runtime/blob/1b80b9629af8ec55ae200e2c649a9603533eed20/pkg/webhook/admission/validator.go#L76

for custom validators object is copied, but not validator
https://github.com/kubernetes-sigs/controller-runtime/blob/1b80b9629af8ec55ae200e2c649a9603533eed20/pkg/webhook/admission/validator_custom.go#L75
